src/fs.register.ts

```diff
@@ -31,9 +32,34 @@ export const fqdn: FinalizeRequestMiddleware<object, MetadataBearer> = (next) =>
   };
 };
 
+/**
+ * AWS SDK middleware logic to try 3 times if receiving an EAI_AGAIN error
+ */
+export function eaiAgainBuilder(timeout: number): BuildMiddleware<object, MetadataBearer> {
+  const eaiAgain: BuildMiddleware<object, MetadataBearer> = (next) => {
+    const maxTries = 3;
+    return async (args) => {
+      for (let i = 0; i < maxTries; i++) {
+        try {
+          return await next(args);
+        } catch (error) {
+          if (error && typeof error === 'object' && 'code' in error && error.code !== 'EAI_AGAIN') {
+            throw error;
+          }
+          logger.error('eai_again:retry:' + i);
+          await setTimeout(timeout);
+        }
+      }
+      throw new Error(`EAI_AGAIN maximum tries (${maxTries}) exceeded`);
+    };
+  };
+  return eaiAgain;
+}
+
 const client = new S3Client();
 export const s3Fs = new FsAwsS3V3(client);
 client.middlewareStack.add(fqdn, { name: 'FQDN', step: 'finalizeRequest' });
+client.middlewareStack.add(eaiAgainBuilder(1000), { name: 'EAI_AGAIN', step: 'build' });
```

Comment:
When seeing a retry handler I generally assume the frist number is the number of retries not the backoff, it is also a good idea to backoff further as the retry count expands.

the S3Client retry handler is configured :

```typescript
new ConfiguredRetryStrategy(
  4, // Retry count
  (attempt: number) => 100 + attempt * 1000 // backoff function.
)
```

could we just copy this interface?